src/resolve/resolved-manager.c

```diff
@@ -357,8 +375,12 @@ static int determine_hostname(char **full_hostname, char **llmnr_hostname, char
                 log_error("System hostname is not UTF-8 clean.");
                 return -EINVAL;
         }
+        decoded = label;
+#else
+        decoded = label; /* no decoding */
+#endif
 
```

Comment:
urks... I really don't like that this spills into all our sources now. Can't we unify the libidn vs. libidn2 arbitration inside of dns_label_undo_idna() so that the callers don't have to know anything about this?